htdocs/editors/CKeditor/formCkeditor.php

```diff
@@ -112,8 +112,54 @@ public function render() {
 
 		$ret = $xoTheme->addScript("/editors/CKeditor/ckeditor/ckeditor.js", array('type' => 'text/javascript'), '');
 		$ret .= $xoTheme->addScript("/editors/CKeditor/ckeditor/adapters/jquery.js", array('type' => 'text/javascript'), '');
-		$ret .= $xoTheme->addScript('', array('type' => 'text/javascript'),
-			'var config = {filebrowserImageBrowseUrl: "' . ICMS_URL . '/editors/CKeditor/imagebrowser.php", toolbar: "' .  $toolbar . '"}; $(function() { $("#'.@$this->_name.'_tarea").ckeditor(config); $("#'.@$this->_name.'_tarea").parents("form").submit(function() { var data = $("#'.@$this->_name.'_tarea").html(); $("#'.@$this->_name.'_tarea").html(data); }); });');
+
+		// CSRF token for AJAX: createToken() already returns the hashed token expected by validateToken()
+		$__tokenVal = icms::$security->createToken(0);
+
+		// Build JS with real newlines to avoid invalid escape sequences in the browser.
+		$textareaId = $this->_name . '_tarea';
+		$icmsUrl = ICMS_URL;
+		$script = <<<JS
```

Comment:
this is bad idea for the future... don't mix big JS parts with PHP code. Better create one extra JS file that has some specvific function, include that file and call that file with parameters

like:

```php
$ret = $xoTheme->addScript("/editors/CKeditor/ckeditor/icms_mentions.js", array('type' => 'text/javascript'), '');
$ret .= $xoTheme->addScript(
   '', 
   array('type' => 'text/javascript'), 
   sprintf(
      'icms_mentions(%s)',
      json_encode(['url' => $icmsUrl, 'textareaId' => $textareaId])
   )
);
```